helpers/lockLoadingFunctions.js

```diff
@@ -0,0 +1,124 @@
+  }
+  /** @type {OriginalLockType} */
+  lock = await OriginalLockType.findByPk(createdLock.OriginalLockType_ID)
+
+  /** @type {LoadedOriginalLock} */
+  deck = await loadOriginalLockType(createdLock)
+
+  deck_ID = deck.Original_Loaded_ID  //Original_Loaded_ID //Original_Loaded_ID
+  params = {
+    Deck_ID: deck_ID,
+    Chance_Period: lock.Chance_Period,
+    Cumulative: lock.Cumulative,
+    Chances: 1,
+    Last_Chance_Time: Date.now(),
+    Hide_Info: lock.Hide_Card_Info,
+    Start_Lock_Frozen: lock.Start_Lock_Frozen, // included for future use
+    Auto_Resets_Enabled: lock.Auto_Resets_Enabled, // included for future use
+    Reset_Frequency: lock.Reset_Frequency, // included for future use
+    Max_Resets: lock.Max_Resets // included for future use
+  }
+  return params
+}
+
+/**
+* Creates either a real or a fake LoadedLock for the user with User_ID based on a CreatedLock and inputs objects.
+* If creating a real lock, is_real_lock should be true, and real_lock_link omitted or undefined.
+* If creating a fake lock, is_real_lock should be fales, and real_lock_link should be the LoadedLock_ID of
+* the corresponding real lock.
+* @param {CreatedLock} createdLock - The CreatedLock from which to create the LoadedLock
+* @param {int} User_ID - the User_ID of the User loading the lock
+* @param {object} inputs - the input object associated with the loadLock mutation
+* @param {boolean} is_real_lock - True if created lock should be real, false if should be fake
+* @param {int?} real_lock_link - The LoadedLock_ID of the corresponding real LoadedLock or undefined 
+* @returns {LoadedLock} - The LoadedLock object created
+*/
+async function createLoadedLock(createdLock, User_ID, inputs, is_real_lock, real_lock_link) {
+  params = (createdLock.OriginalLockType_ID) ? // true if variable, false if fixed
+    await findVariableProperties(createdLock) :
+    await findFixedProperties(createdLock)
+  const loadedLock = await LoadedLock.create({
+    CreatedLock_ID: createdLock.Lock_ID,
+    Lockee: User_ID,
+    Keyholder: createdLock.User_ID,
+    Code: inputs.Code || await NewCode(User_ID), //tested
+    Original_Lock_Deck: params.Deck_ID,
+    Timed_Unlock_Time: params.Timed_Unlock_Time,
+    Hide_Info: params.Hide_Info,
+    Emergency_Keys_Enabled: inputs.Emergency_Keys,
+    Emergency_Keys_Amount: inputs.Emergency_Keys_Amount, 
+    Test_Lock: inputs.Test_Lock,
+    Trusted: inputs.Trust_Keyholder,
+    Cumulative: params.Cumulative,
+    Chance_Period: params.Chance_Period,
+    Chances: params.Chances,
+    //Last_Pick_Time not initialized, so NULL until first pick
+    Last_Chance_Time: params.Last_Chance_Time,
+    // Current_Freeze_ID: not initialized, so starts as NULL.  Added below
+    // Lockee_Rating, KeyholderRating  - left NULL at start of lock
+    Unlocked: false,
+    Free_Unlock: false,
+    Fake_Lock: !is_real_lock,
+    Real_Lock: real_lock_link
```

Comment:
This is true.  If it's a real lock, then Fake_Lock is false, and Real_Lock is NULL, while if it's a fake lock, then Fake_Lock is true, and Real_Lock has a link to the related real lock.  We could refactor this to have just the Real_Lock field and use that to determine whether the lock is real. It might be best to rename the field to something such as Real_Lock_ID so that it's more obvious what is stored in that field.

Reply:
I'm changing the code here as well as the models, but not the migration.  I'm also changing the schema of LoadedLock to remove the Fake_Lock field and to add the Real_Lock field, since the keyholder view should disclose this information.  Eventually we need a Lockee view of LoadedLock that doesn't disclose the code or whether lock is real.